mbf_abstract_nav/src/abstract_planner_execution.cpp

```diff
@@ -312,43 +308,33 @@ void AbstractPlannerExecution::run()
           // old nav_core-based planners do not support canceling), and we add here the fact to the log for info
           ROS_INFO_STREAM("Planning patience (" << patience_.toSec() << "s) has been exceeded"
                                                 << (cancel_ ? "; planner canceled!" : ""));
-          exceeded = true;
-          planning_ = false;
           setState(PAT_EXCEEDED, true);
         }
         else if (max_retries_ == 0 && patience_.isZero())
         {
           ROS_INFO_STREAM("Planning could not find a plan!");
-          exceeded = true;
-          planning_ = false;
           setState(NO_PLAN_FOUND, true);
         }
         else
         {
-          exceeded = false;
+          planning_ = true;
           ROS_DEBUG_STREAM("Planning could not find a plan! Trying again...");
         }
       }
-      else if (cancel_)
-      {
-        ROS_INFO_STREAM("The global planner has been canceled!");
-        planning_ = false;
-        setState(CANCELED, true);
-      }
     } // while (planning_ && ros::ok())
   }
   catch (const boost::thread_interrupted &ex)
   {
     // Planner thread interrupted; probably we have exceeded planner patience
     ROS_WARN_STREAM("Planner thread interrupted!");
-    planning_ = false;
     setState(STOPPED, true);
   }
   catch (...)
   {
     ROS_FATAL_STREAM("Unknown error occurred: " << boost::current_exception_diagnostic_information());
     setState(INTERNAL_ERROR, true);
   }
+  planning_ = false;
```

Comment:
if the planner would throw, we would miss to reset the planning_ to false (then start() would always return false)